update_repo.sh

```diff
@@ -50,30 +68,40 @@ check_repo()
     fi
 }
 
-
 update_repo()
 {
-    OPTS=
-    [ -r "$1/comps.xml" ] && OPTS="-g comps.xml"
-    $createrepo $OPTS --update -x 'fc*/*' -o $1 $1 >/dev/null
-    if [ -x "$mkmetalink" -a -r "$mirrors_list" ]; then
-        $mkmetalink $mirrors_list "$1/repodata/repomd.xml" \
-            > "$1/repodata/repomd.xml.metalink"
+    local OPTS= tosign sign_key
+
+    # If we are told to not sign anything, we don’t bother generating metadata.
+    # Unsigned metadata will be rejected anyway, so generating it is wasteful.
+    # Furthermore, it risks clobbering signed metadata that already exists.
+    if [ "x$NO_SIGN" != x1 ]; then
+        if [ "x${SIGN_KEY+test}" = xtest ]; then
+            sign_key=$SIGN_KEY
+        else
+            sign_key=$(rpm --eval '%{?_gpg_name:%_gpg_name}') || return
+        fi
+        if [ -n "$sign_key" ]; then
+            if [ -z "${GNUPG-}" ]; then
+                GNUPG=$(rpm --eval '%{?__gpg:%__gpg}') || return
+            fi
+            [ -r "$1/comps.xml" ] && OPTS="-g comps.xml"
+            "$createrepo" $OPTS --update -x 'fc*/*' -o "$1" -- "$1" >/dev/null || return
+            "$GNUPG" --detach-sign --armor "--local-user=$sign_key" \
```

Comment:
This needs `--output=-` to work with standard gpg (mostly for testing purposes).